server/src/main/java/org/apache/iotdb/db/qp/strategy/optimizer/ConcatPathOptimizer.java

```diff
@@ -292,23 +294,39 @@ private FilterOperator constructBinaryFilterTreeWithAnd(
         currentNode.addChildOperator(newInnerNode);
         currentNode = newInnerNode;
       }
-      try {
-        if (operator instanceof InOperator) {
+      switch (operator.getType()) {
+        case IN:
           currentNode.addChildOperator(
               new InOperator(
                   operator.getTokenIntType(),
                   noStarPaths.get(i),
                   ((InOperator) operator).getNot(),
                   ((InOperator) operator).getValues()));
-        } else {
+          break;
+        case LIKE:
           currentNode.addChildOperator(
-              new BasicFunctionOperator(
+              new LikeOperator(
                   operator.getTokenIntType(),
                   noStarPaths.get(i),
-                  ((BasicFunctionOperator) operator).getValue()));
-        }
-      } catch (SQLParserException e) {
-        throw new LogicalOptimizeException(e.getMessage());
+                  ((LikeOperator) operator).getValue()));
+          break;
+        case REGEXP:
+          currentNode.addChildOperator(
+              new RegexpOperator(
+                  operator.getTokenIntType(),
+                  noStarPaths.get(i),
+                  ((RegexpOperator) operator).getValue()));
+          break;
+        default:
+          try {
+            currentNode.addChildOperator(
+                new BasicFunctionOperator(
+                    operator.getTokenIntType(),
+                    noStarPaths.get(i),
+                    ((BasicFunctionOperator) operator).getValue()));
+          } catch (SQLParserException e) {
+            throw new LogicalOptimizeException(e.getMessage());
+          }
       }
```

Comment:
Hi, is it OK to try catch SQLParserException for BasicFunctionOperator only?

Reply:
Other Operators are created without exceptions, the default is BasicFunctionOperator may have exceptions